### PR TITLE
Add plugin for dnf (New package manager and package management librar…Add plugin for dnf (New package manager and package management library for Fedora) based in yum plugin

### PR DESCRIPTION
**Pull request by fjospinas**
Add plugin for dnf (New package manager and package management library for Fedora) based in yum plugin.

DNF page: https://fedoraproject.org/wiki/Features/DNF

### CODE REVIEW

**Comment by ohadlevy** [comment]
@fjospinas would you add auto complete too? thanks!

**Comment by sebastianbergmann** [comment]
This is basically the same as #2452 with the only differences being that `sudo dnf upgrade` instead of `sudo dnf update" and `sudo dnf remove` instead of `sudo dnf erase` being used.

According to `dnf --help`, the commands used for the `dnfu` and `dnfr` aliased in this pull request are the correct/new/current ones:

    $ dnf --version
    1.0.0
      Installed: dnf-0:1.0.0-1.fc22.noarch at 2015-05-21 18:56
      Built    : Fedora Project at 2015-05-02 13:00

      Installed: rpm-0:4.12.0.1-9.fc22.x86_64 at 2015-05-21 18:56
      Built    : Fedora Project at 2015-04-15 09:21


    $ dnf --help
    usage: dnf [options] COMMAND

    List of Main Commands

    autoremove                
    check-update              Check for available package upgrades
    clean                     Remove cached data
    distro-sync               Synchronize installed packages to the latest available versions
    downgrade                 downgrade a package
    group                     Display, or use, the groups information
    help                      Display a helpful usage message
    history                   Display, or use, the transaction history
    info                      Display details about a package or group of packages
    install                   Install a package or packages on your system
    list                      List a package or groups of packages
    makecache                 Generate the metadata cache
    provides                  Find what package provides the given value
    reinstall                 reinstall a package
    remove                    Remove a package or packages from your system
    repolist                  Display the configured software repositories
    repository-packages       Run commands on top of all packages in given repository
    search                    Search package details for the given string
    updateinfo                Display advisories about packages
    upgrade                   Upgrade a package or packages on your system
    upgrade-to                Upgrade a package on your system to the specified version

    List of Plugin Commands

    builddep                  Install build dependencies for packagei or spec file
    config-manager            manage dnf configuration options and repositories
    copr                      Interact with Copr repositories.
    debuginfo-install         install debuginfo packages
    download                  Download package to current directory
    kickstart                 Install packages defined in a kickstart file on your system
    migrate                   migrate yum's history, group and yumdb data to dnf
    needs-restarting          determine updated binaries that need restarting
    playground                Interact with Playground repository.
    repoquery                 search for packages matching keyword
    reposync                  download all packages from remote repo

    optional arguments:
      --allowerasing        allow erasing of installed packages to resolve
                            dependencies
      -b, --best            try the best available package versions in
                            transactions.
      -C, --cacheonly       run entirely from system cache, don't update cache
      -c [config file], --config [config file]
                            config file location
      -d [debug level], --debuglevel [debug level]
                            debugging output level
      --debugsolver         dumps detailed solving results into files
      --showduplicates      show duplicates, in repos, in list/search commands
      -e ERRORLEVEL, --errorlevel ERRORLEVEL
                            error output level
      --rpmverbosity [debug level name]
                            debugging output level for rpm
      -q, --quiet           quiet operation
      -v, --verbose         verbose operation
      -y, --assumeyes       answer yes for all questions
      --assumeno            answer no for all questions
      --version             show DNF version and exit
      --installroot [path]  set install root
      --enablerepo [repo]
      --disablerepo [repo]
      -x [package], --exclude [package]
                            exclude packages by name or glob
      --disableexcludes [repo]
                            disable excludes
      --obsoletes           enable obsoletes processing during upgrades
      --noplugins           disable all plugins
      --nogpgcheck          disable gpg signature checking
      --disableplugin [plugin]
                            disable plugins by name
      --color COLOR         control whether color is used
      --releasever RELEASEVER
                            override the value of $releasever in config and repo
                            files
      --setopt SETOPTS      set arbitrary config and repo options
      --refresh             set metadata as expired before running the command
      -4                    resolve to IPv4 addresses only
      -6                    resolve to IPv6 addresses only
      -h, --help            show help

**Comment by sebastianbergmann** [comment]
@ohadlevy While I agree that support for auto-completion of package names for aliases such as `dnfi` is desirable I think that it is outside the scope of this pull request which is about adding a plugin for dnf. The old yum plugin did not have support for auto-completion either.

File: plugins/dnf/dnf.plugin.zsh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+alias dnfgl="dnf grouplist"                   # list package groups
+alias dnfli="dnf list installed"              # print all installed packages
+alias dnfmc="dnf makecache"                   # rebuilds the dnf package list
+
+alias dnfu="sudo dnf upgrade"                  # upgrate packages
+alias dnfi="sudo dnf install"                 # install package
+alias dnfgi="sudo dnf groupinstall"           # install package group
+alias dnfr="sudo dnf remove"                  # remove package
+alias dnfgr="sudo dnf groupremove"            # remove pagage group
+alias dnfc="sudo dnf clean all"               # clean cache
```
Comment (sebastianbergmann) [inline plugins/dnf/dnf.plugin.zsh]: File does not end with a single blank line.

File: plugins/dnf/dnf.plugin.zsh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+## Aliases
+
+alias dnfs="dnf search"                       # search package
+alias dnfp="dnf info"                         # show package info
+alias dnfl="dnf list"                         # list packages
+alias dnfgl="dnf grouplist"                   # list package groups
+alias dnfli="dnf list installed"              # print all installed packages
+alias dnfmc="dnf makecache"                   # rebuilds the dnf package list
+
+alias dnfu="sudo dnf upgrade"                  # upgrate packages
```
Comment (sebastianbergmann) [inline plugins/dnf/dnf.plugin.zsh]: * Comment is not aligned with other comments
* `upgrate` should be `upgrade`

**Pull request closed without merging**